src/dialog.rs

```diff
@@ -0,0 +1,46 @@
+extern crate ole32;
+extern crate uuid;
+extern crate winapi;
+
+use winapi::*;
+use std::ptr::null_mut;
+use util::{FromWide};
+
+pub unsafe fn get_open_file_dialog_path(hwnd_owner: HWND) -> Option<String> {
+  get_file_dialog_path(hwnd_owner, true)
+}
+
+pub unsafe fn get_save_file_dialog_path(hwnd_owner: HWND) -> Option<String> {
+  get_file_dialog_path(hwnd_owner, false)
+}
+
+unsafe fn get_file_dialog_path(hwnd_owner: HWND, open: bool) -> Option<String> {
```

Comment:
I decided to use a `bool` here even though it's poor modeling because I need to pass two pieces of info - the class and the ID, the they are both either `Open` or `Save` so I wanted to prevent an invalid invariant. This is then a two-state value but I felt creating an `enum` for this might have been an overkill given the two call sites in the same file? Not sure.

Reply:
It's reasonable.